### ISSUE TITLE
Error: Failed to upgrade vendor ruby [ARM]

### ISSUE THREAD

**Issue by vikas12121200** [Linuxbrew/brew#781]
```
pi@raspberrypi:~ $ sh -c "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)"
==> This script will install:
/home/linuxbrew/.linuxbrew/bin/brew
/home/linuxbrew/.linuxbrew/share/doc/homebrew
/home/linuxbrew/.linuxbrew/share/man/man1/brew.1
/home/linuxbrew/.linuxbrew/share/zsh/site-functions/_brew
/home/linuxbrew/.linuxbrew/etc/bash_completion.d/brew
/home/pi/.cache/Homebrew/
/home/linuxbrew/.linuxbrew/Homebrew

Press RETURN to continue or any other key to abort
==> Downloading and installing Linuxbrew...
HEAD is now at 4ff4e7f32 update-report: Work around an unusual error (#779)
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.sh: line 4: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.sh: line 4: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.sh: line 4: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
==> Downloading https://homebrew.bintray.com/bottles-portable-ruby/portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
Already downloaded: /home/pi/.cache/Homebrew/portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = "en_US.UTF-8",
	LANG = (unset)
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
==> Pouring portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
Error: Failed to upgrade vendor ruby.
Error: Failed to install vendor Ruby.
Failed during: /home/linuxbrew/.linuxbrew/bin/brew update --force
```

**Comment by vikas12121200** [comment]
the path is set with this command
sudo localedef -i en_US -f UTF-8 en_US.UTF-8

but this error remain
 Pouring portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
Error: Failed to upgrade vendor ruby.
Error: Failed to install vendor Ruby.

**Comment by sjackman** [comment]
Please surround copied-and-pasted logs with triple back ticks. See [GitHub Help / Quoting Code](https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code).

**Comment by sjackman** [comment]
Please report the output of…
```sh
/home/linuxbrew/.linuxbrew/bin/brew update-reset
/home/linuxbrew/.linuxbrew/bin/brew vendor-install ruby
```

**Comment by vikas12121200** [comment]
```
pi@raspberrypi:~ $ /home/linuxbrew/.linuxbrew/bin/brew update-reset
==> Fetching /home/linuxbrew/.linuxbrew/Homebrew...
remote: Counting objects: 278, done.
remote: Total 278 (delta 1), reused 1 (delta 1), pack-reused 277
Receiving objects: 100% (278/278), 221.46 KiB | 74.00 KiB/s, done.
Resolving deltas: 100% (144/144), completed with 1 local object.
From https://github.com/Linuxbrew/brew
 * [new branch]          gh-pages   -> origin/gh-pages

==> Resetting /home/linuxbrew/.linuxbrew/Homebrew...
Branch master set up to track remote branch master from origin.
Reset branch 'master'
Your branch is up-to-date with 'origin/master'.
```

**Comment by vikas12121200** [comment]
```
pi@raspberrypi:~ $ /home/linuxbrew/.linuxbrew/bin/brew vendor-install ruby
==> Downloading https://homebrew.bintray.com/bottles-portable-ruby/portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
Already downloaded: /home/pi/.cache/Homebrew/portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
==> Pouring portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
Error: Failed to upgrade vendor `ruby.```
```

**Comment by sjackman** [comment]
`portable-ruby` 2.3.7 needs to be built for ARM. I'm working on that now.
See https://github.com/Linuxbrew/brew/blob/4ff4e7f32fcb854b334c529351a8bb8133dca674/Library/Homebrew/cmd/vendor-install.sh#L29

**Comment by vikas12121200** [comment]
so sir when it will be available

**Comment by vikas12121200** [comment]
and what to do now please tell me

**Comment by sjackman** [comment]
I tried building `portable-ruby` 2.3.7 for ARM just now, and ran into an issue. Hopefully within a week if I'm able to resolve this problem.

In the mean time, try editing `~/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby-version` and change `2.3.7` to `2.3.3_2`.

**Comment by vikas12121200** [comment]
its working but error remain
```
pi@raspberrypi:~ $ brew install hello
Updating Homebrew...
To restore the stashed changes to /home/linuxbrew/.linuxbrew/Homebrew run:
  'cd /home/linuxbrew/.linuxbrew/Homebrew && git stash pop'
==> Downloading https://homebrew.bintray.com/bottles-portable-ruby/portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
Already downloaded: /home/pi/.cache/Homebrew/portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
==> Pouring portable-ruby-2.3.3_2.armv6_linux.bottle.tar.gz
Error: Failed to upgrade vendor ruby.
Error: Failed to upgrade vendor Ruby.
==> Tapping homebrew/core
Cloning into '/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Counting objects: 4859, done.
remote: Compressing objects: 100% (4630/4630), done.
remote: Total 4859 (delta 45), reused 4285 (delta 37), pack-reused 0
Receiving objects: 100% (4859/4859), 4.09 MiB | 412.00 KiB/s, done.
Resolving deltas: 100% (45/45), done.
Tapped 3 commands and 4640 formulae (4,900 files, 12.9MB).
==> Downloading https://ftp.gnu.org/gnu/hello/hello-2.10.tar.gz
######################################################################## 100.0%
==> ./configure --disable-silent-rules --prefix=/home/linuxbrew/.linuxbrew/Cellar/hello/2.10
==> make install
🍺  /home/linuxbrew/.linuxbrew/Cellar/hello/2.10: 53 files, 646.7KB, built in 2 minutes 26 seconds
```

**Comment by sjackman** [comment]
Okay. Glad to hear that this workaround got you further along. I'll work on the proper fix.

**Comment by sjackman** [comment]
I've run into trouble building `portable-ruby` for ARM. It builds fine, except for the `dbm` module, which doesn't build. `brew test portable-ruby` fails because this module is not built.

**Comment by sjackman** [comment]
Excellent news on this front! Homebrew/brew no longer uses the Ruby `dbm` module, so I'll be able to build a `portable-ruby` bottle for ARM without it. See https://github.com/Homebrew/brew/pull/4984 `cache_store: use JSON.`

**Comment by sjackman** [comment]
I've built the `portable-ruby` bottle for ARM! See PR https://github.com/Linuxbrew/brew/pull/859
This issue will be fixed in Linuxbrew 1.7.7. If you'd like to test it in the mean time, run
```sh
HOMEBREW_DEVELOPER=1 brew update
```
